src/CacheTagsInvalidator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\cloudflare\CacheTagsInvalidator.
+ */
+
+namespace Drupal\cloudflare;
+
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use CloudFlarePhpSdk\Exceptions\CloudFlareHttpException;
+use CloudFlarePhpSdk\Exceptions\CloudFlareApiException;
+use Drupal\cloudflare\EventSubscriber\CloudFlareCacheTagHeaderGenerator;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
+
+/**
+ * Cache tags invalidator implementation that invalidates CloudFlare.
+ */
+class CacheTagsInvalidator implements CacheTagsInvalidatorInterface {
```

Comment:
I wonder if this invalidator should be in a separate module? Especially since this will not work with Varnish.  However, I don't feel strongly. We can remove when purge is implemented.